lumicks/pylake/kymo.py

```diff
@@ -570,6 +572,74 @@ def plot_with_time_histogram(
         ax_hist.set_ylabel("counts")
         ax_hist.set_title(self.name)
 
+    def bead_roi(self, bead_diameter, *, channel="green", plot=False, percentile=70, **kwargs):
+        """Determine approximate positional coordinates of the bead edges.
+
+        Searches for the bead edges by checking where the fluorescence drops below a threshold.
+        Only intended for non-moving beads.
+
+        .. warning::
+
+            This is early access alpha functionality. While usable, this has not yet been tested in
+            a large number of different scenarios. The API can still be subject to change without
+            any prior deprecation notice! If you use this functionality keep a close eye on the
+            changelog for any changes that may affect your analysis.
+
+        Parameters
+        ----------
+        bead_diameter : float
+            Rough estimate for the bead size (microns).
+        channel : 'red', 'green', 'blue', optional
+            Channel to use for bead detection.
+        plot : bool, optional
+            Plot result
+        percentile : int
+            Percentile to drop down to before accepting that we have left the bead area. Higher
+            values will make the bounds go closer to the bead edge but risk the algorithm failing
+            or having high background.
+        **kwargs
+            Forwarded to bead finding function.
+
+        Raises
+        ------
+        RuntimeError
+            When the algorithm fails to locate two edges during the bead finding stage.
+        """
+
+        return (
+            find_beads(
+                self.get_image(channel).sum(axis=1),
+                bead_diameter_pixels=bead_diameter / self.pixelsize_um[0],
+                plot=plot,
+                percentile=percentile,
+                **kwargs,
+            )
+            * self.pixelsize_um[0]
+        )
+
+    def crop_beads(self, bead_diameter, channel="green", **kwargs):
+        """Crop beads out of kymograph
+
+        .. warning::
+
+            This is early access alpha functionality. While usable, this has not yet been tested in
+            a large number of different scenarios. The API can still be subject to change without
+            any prior deprecation notice! If you use this functionality keep a close eye on the
+            changelog for any changes that may affect your analysis.
+
+        Parameters
+        ----------
+        bead_diameter : float
+            Rough estimate for the bead size (microns).
+        channel : 'red', 'green', 'blue', optional
+            Channel to use for bead detection.
+        **kwargs
+            Forwarded to :meth:`Kymo.bead_roi()`.
+        """
+        to_current_units = self.pixelsize[0] / self.pixelsize_um[0]
+        crop_locations = np.asarray(self.bead_roi(bead_diameter, channel=channel, **kwargs))
```

Comment:
In a way it's nice that you take care to handle any units (although the docstring specifically mentions microns which could be confusing since it needs to be in whatever the current units are). I wonder though if we should just `raise` for this and the previous method if the kymo isn't in microns? It doesn't make sense really to calibrate and then crop, and there's less chance of some odd corner case popping up that we haven't taken into account (which is just a nice way of saying it's less for us to worry about 😉 ) What do you think?

Reply:
Yeah, probably a good idea to explicitly exclude this.